File: fixtures/projects/e2e-python/src/core/scheduler.py
from dataclasses import dataclass

from src.core.engine import EngineState, load_engine_state
from src.game.player import Player, create_player


@dataclass(slots=True)
class Scheduler:
    state: EngineState
    owner: Player


def build_scheduler(players: list[dict[str, str]]) -> Scheduler:
    state = load_engine_state(players)
    owner = create_player(state.players[0])
    return Scheduler(state=state, owner=owner)

File: fixtures/projects/e2e-ruby/lib/core/scheduler.rb
require_relative "./engine.rb"
require_relative "../game/player.rb"

module Core
  class Scheduler
    def run(job)
      {
        engine: Engine.new.boot(user_id: job[:user_id], name: job[:name]),
        player: Game::Player.new.build_profile(job[:user_id], job[:name]),
      }
    end
  end
end
